test/listen_SUITE.erl

```diff
@@ -61,10 +105,88 @@ already_test(Config) ->
 
     ok.
 
-bad_addr_test(Config) ->
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+bad_addr(Config) ->
     Swarm = proplists:get_value(swarm, Config),
 
     {error, {unsupported_address, _}} = libp2p_swarm:listen(Swarm, "/onion/timaq4ygg2iegci7:1234"),
     {error, {unsupported_address, _}} = libp2p_swarm:listen(Swarm, "/udp/1234/udt"),
 
     ok.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+port0_reuse(_Config) ->
+    SwarmOpts = [{libp2p_transport_tcp, [{nat, false}]}],
+
+    {ok, Swarm} = libp2p_swarm:start(listen_port0_reuse, SwarmOpts),
+    ok = libp2p_swarm:listen(Swarm, "/ip4/0.0.0.0/tcp/0"),
+    ok = libp2p_swarm:listen(Swarm, "/ip6/::/tcp/0"),
+
+    timer:sleep(2000),
+    ListenAddrs = libp2p_swarm:listen_addrs(Swarm),
+
+    ok = libp2p_swarm:stop(Swarm),
+    timer:sleep(2000),
```

Comment:
Ditto? Is this assuming how long it takes to tear down the swarm? It seems very brittle?